include/deal.II/lac/scalapack.h

```diff
@@ -0,0 +1,501 @@
+  void eigenvalues_symmetric (std::vector<NumberType> &eigenvalues);
+
+  /**
+   * Compute all eigenpairs of real symmetric matrix using pXsyev.
+   * After successful computation the eigenvalues are returned in @p eigenvalues
+   * in ascending order. The eigenvectors are stored in the columns of the matrix,
+   * therefore overwriting the original content of the matrix.
+   */
+  void eigenpairs_symmetric (std::vector<NumberType> &eigenvalues);
+
+  /**
+   * Estimate the the condition number of a SPD matrix in $l_1$-norm.
+   * The matrix has to be in Cholesky state. The reciprocal of the
+   * condition number is returned in order to avoid the possibility of
+   * overflow when the condition number is very large.
+   *
+   * @p a_norm shall contain $l_1$-norm of the matrix prior to calling
+   * Cholesky factorization.
+   *
+   * @note an alternative is to compute the inverse of the matrix
+   * explicitly and manually constructor $k_1 = ||A||_1 ||A^{-1}||_1$.
+   */
+  NumberType reciprocal_condition_number(const NumberType a_norm) const;
+
+  /**
+   * Compute $l_1$-norm of the matrix.
+   * @return
+   */
+  NumberType l1_norm() const;
+
+  /**
+   * Compute $l_{\infty}$ norm of the matrix.
+   */
+  NumberType linfty_norm() const;
+
+  /**
+   * Compute the Frobenius norm of the matrix.
+   */
+  NumberType frobenius_norm() const;
+
+  /**
+   * Number of rows of the matrix $M \times N$.
+   */
+  int m() const;
+
+  /**
+   * Number of columns of the matrix $M \times N$.
+   */
+  int n() const;
+
+  /**
+   * Number of local rows on this MPI processes.
+   */
+  int local_m() const;
+
+  /**
+   * Number of local columns on this MPI process.
+   */
+  int local_n() const;
+
+  /**
+   * Return global row number for the given local row @p loc_row .
+   */
+  int global_row(const int loc_row) const;
+
+  /**
+   * Return global column number for the given local column @p loc_column.
+   */
+  int global_column(const int loc_column) const;
+
+  /**
+   * Read access to local element.
+   */
+  NumberType local_el(const int loc_row, const int loc_column) const;
+
+  /**
+   * Write access to local element.
+   */
+  NumberType &local_el(const int loc_row, const int loc_column);
+
+private:
+
+  /**
+   * Calculate norm of a distributed dense matrix using ScaLAPACK's
+   * internal function.
+   */
+  NumberType norm(const char type) const;
+
+  /**
+   * Send @p value from process with rank zero to processes which
+   * are not in the process grid.
+   */
+  void send_to_inactive(NumberType &value) const;
+
+  /**
+   * Since ScaLAPACK operations notoriously change the meaning of the matrix
+   * entries, we record the current state after the last operation here.
+   */
+  LAPACKSupport::State state;
+
+  /**
+   * Additional property of the matrix which may help to select more
+   * efficient ScaLAPACK functions.
+   */
+  LAPACKSupport::Property property;
+
+  /**
+   * Shared pointer to ProcessGrid object which contains Blacs context
```

Comment:
A shared pointer to a ProcessGrid object that contains a BLACS context and a MPI communication, as well as